rclcpp/include/rclcpp/callback_group.hpp

```diff
@@ -61,6 +62,10 @@ class CallbackGroup
   RCLCPP_PUBLIC
   explicit CallbackGroup(CallbackGroupType group_type);
 
+  RCLCPP_PUBLIC
+  const std::vector<rclcpp::PublisherBase::WeakPtr> &
```

Comment:
Passing std::vector through public interfaces is not ideal IMHO - I understand why it's done there, and maybe this is not the first occurrence here but we should be super careful (see your problems in clang libstdcxx for a good reason to care :) )

`get_publishers_count()` + `get_publisher(int i)` is a way to avoid this issue maybe?